test/caption-stream.test.js

```diff
@@ -2735,6 +2761,69 @@ QUnit.test('parses 708 captions', function(assert) {
   }, 'parsed caption 234 correctly');
 });
 
+QUnit.test('Decodes multibyte characters if valid encoding option is provided and TextDecoder is supported', function(assert) {
+  var captions = [];
+
+  cea708Stream = new m2ts.Cea708Stream({
+    captionServices: {
+      SERVICE1: {
+        encoding: 'euc-kr'
+      }
+    }
+  });
+
+  cea708Stream.on('data', function(caption) {
+    captions.push(caption);
+  });
+
+  cc708Korean.forEach(cea708Stream.push, cea708Stream);
+
+  cea708Stream.flushDisplayed(4721138662, cea708Stream.services[1]);
+
+  assert.equal(captions.length, 1, 'parsed single caption correctly');
+
+  if (window.TextDecoder) {
+    assert.ok(cea708Stream.services[1].textDecoder_, 'TextDecoder created when supported');
+    assert.equal(
+      captions[0].text,
+      '니가 ',
+      'parsed multibyte characters correctly'
+    );
+  } else {
+    assert.notOk(cea708Stream.services[1].textDecoder_, 'TextDecoder not created when unsupported');
+  }
+});
+
+QUnit.test('Creates TextDecoder only if valid encoding value is provided', function(assert) {
+  var secondCea708Stream;
+
+  cea708Stream = new m2ts.Cea708Stream({
+    captionServices: {
+      SERVICE1: {
+        encoding: 'euc-kr'
+      }
+    }
+  });
+
+  cc708Korean.forEach(cea708Stream.push, cea708Stream);
+  cea708Stream.flushDisplayed(4721138662, cea708Stream.services[1]);
+
+  assert.ok(cea708Stream.services[1].textDecoder_, 'TextDecoder created successfully when encoding is valid');
+
+  secondCea708Stream = new m2ts.Cea708Stream({
+    captionServices: {
+      SERVICE1: {
+        encoding: 'invalid'
+      }
+    }
+  });
+
+  cc708Korean.forEach(secondCea708Stream.push, secondCea708Stream);
+  secondCea708Stream.flushDisplayed(4721138662, secondCea708Stream.services[1]);
+
+  assert.notOk(secondCea708Stream.services[1].textDecoder_, 'TextDecoder not created when encoding is invalid');
+});
```

Comment:
this test needs to check for IE11 or if TextDecoder is present.

Reply:
Added a check